tests/test_databases.py

```diff
@@ -0,0 +1,64 @@
+import ase
+
+
+def test_cod_query_widget():
+    """Test the COD query widget."""
+    from aiidalab_widgets_base import CodQueryWidget
+
+    widget = CodQueryWidget()
+
+    # Enter the query string.
+    widget.inp_elements.value = "Ni Ti"
+
+    # Run the query.
+    widget._on_click_query()
+
+    # Select on of the results.
+    widget.drop_structure.label = "NiTi (id: 1100132)"
+
+    # Check that the structure was loaded.
+    assert isinstance(widget.structure, ase.Atoms)
+    assert widget.structure.get_chemical_formula() == "NiTi"
+
+
+def test_optimade_query_widget():
+    """Test the OPTIMADE query widget."""
+    from aiidalab_widgets_base import OptimadeQueryWidget
+
+    widget = OptimadeQueryWidget()
+
+    # At the present state I cannot check much. Most of the variables are locals of the  __init__ method.
+
+    assert widget.structure is None
+
+
+def test_computational_resources_database_widget():
+    """Test the structure browser widget."""
+    from aiidalab_widgets_base.databases import ComputationalResourcesDatabaseWidget
+
+    # Initiate the widget with no arguments.
+    widget = ComputationalResourcesDatabaseWidget()
+    assert "merlin.psi.ch" in widget.database
+
+    # Initialize the widget with default_calc_job_plugin="cp2k"
+    widget = ComputationalResourcesDatabaseWidget(default_calc_job_plugin="cp2k")
+    assert (
+        "merlin.psi.ch" not in widget.database
+    )  # Merlin does not have CP2K installed.
```

Comment:
You could maybe instead select a non-existing plugin instead of cp2k and assert that widget.database is empty (this happens for example for `orca` currently). And for cp2k just assert that it is non-empty.